compositor_render/src/transformations/shader/validation/error.rs

```diff
@@ -6,6 +10,34 @@ use crate::{
 const HEADER_DOCS_URL: &str =
     "https://github.com/membraneframework/video_compositor/wiki/Shader#header";
 
+#[derive(Debug, thiserror::Error)]
+pub struct ShaderParseError {
+    parse_error: naga::front::wgsl::ParseError,
+    location: Option<SourceLocation>,
+}
+
+impl ShaderParseError {
+    pub fn new(parse_error: naga::front::wgsl::ParseError, source: &str) -> Self {
+        let location = parse_error.location(source);
```

Comment:
I think it would be better to move this line to the `Display` implementation. Keeping this value in the `ShaderParseError` makes us have two sources of truth.